src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/NewAzureCognitiveServicesAccount.cs

```diff
@@ -155,6 +155,12 @@ public string KeyVaultUri
         [AllowEmptyCollection]
         public PSNetworkRuleSet NetworkRuleSet { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The network access type for Cognitive Services Account.")]
+        [ValidateNotNull]
```

Comment:
please add
[ValidateSet("Enabled", "Disabled", IgnoreCase = true)]

and also add possible values to help message

Reply:
Hi @VeryEarly 
We were suggested (by network team) not to limited the values to "Enabled",  "Disabled". Because they might want to add values in future. with this background, do you still suggest to add ValidateSet?

Reply:
it is defined in [swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json#L975), that means if there are more values in the future it won't be a silently update. SDK need to be regenerated, PS models need to be updataed as well.

So I recommend put value set and help messages here as the name "PublicNetworkAccess" is not self explained, and you can still add more values into set in the future.

Reply:
Yes its already changed to what you recommende.